### PR TITLE
2016-03 Aquifer and Build Systems

### PR DESCRIPTION
**Pull request by tsmith512**
This is a work-in-progress PR so that I can collect some input on a timely post.

View the rendered markdown at: https://github.com/tsmith512/blog-posts/blob/2016-03-aquifer-and-build-systems/2013-03-aquifer-and-build-systems/README.md or just check the _Files Changed_ tab since there is no media.

### CODE REVIEW

File: 2013-03-aquifer-and-build-systems/README.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+# Aquifer and Build Systems
+
+## What is a build system?
+
+Build systems simplify the development process by leveraging a single tool to
+handle dependency management, automate tasks like testing, consolidate steps
+to compile/deploy the product, and [????].
```
Comment (milsyobtaf) [inline 2013-03-aquifer-and-build-systems/README.md]: Simplifying onboarding?

File: 2013-03-aquifer-and-build-systems/README.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+**New way:** [Aquifer Coder][AQCDR] can run lint all PHP and JavaScript in your
+repo (which is only your own code) with `aquifer lint` to ensure adherence to
+Drupal coding style and standards for better consistency and code sharing.
+
+Additionally, use [Aquifer Run][AQR] to execute post-build commands to prepare
+your frontend assets, execute tests, or any other repetitive tasks. In the
+future, there will be dedicated Aquifer extensions for different testing
+frameworks or frontend toolchains, but using Run in the meantime reduces the
+number of manual steps.
+
```
Comment (tsmith512) [inline 2013-03-aquifer-and-build-systems/README.md]: Are there more? I'm focused on highlight/high-value features, I know Aquifer provides a ton of things that I'm not listing, but is there anything that jumps out?

File: 2013-03-aquifer-and-build-systems/README.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+handle dependency management, automate tasks like testing, consolidate steps
+to compile/deploy the product, and [????].
+
+Using a build system often leads to using an organized directory and file
+structure which can be reused across projects, reinforcing best practices,
+simplifying the process to ramp up new teammates, and making maintainable
+choices [_(StackOverflow by haylem)_][BBTSO].
+
+While build systems have been in use in software projects for decades, they're
+newer for web projects. Web development used to be serving static assets (HTML,
```
Comment (milsyobtaf) [inline 2013-03-aquifer-and-build-systems/README.md]: used to be [limited to] serving static assets

File: 2013-03-aquifer-and-build-systems/README.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+> Aquifer is a command line interface that makes it easy to scaffold, build,
+> test, and deploy your Drupal websites. It provides a default set of tools that
+> allow you to develop, and build Drupal sites using the Drush-make workflow. In
+> addition, Aquifer ships with an extensions system that allows you to add
+> additional tools to your project.
+> 
+> - [Aquifer.io][AQ]
+
+At Four Kitchens, we saw several pain points in developing and maintain large
```
Comment (stardlc) [inline 2013-03-aquifer-and-build-systems/README.md]: 'maintaining' rather than 'maintain'

File: 2013-03-aquifer-and-build-systems/README.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+## What is a build system?
+
+Build systems simplify the development process by leveraging a single tool to
+handle dependency management, automate tasks like testing, consolidate steps
+to compile/deploy the product, and [????].
+
+Using a build system often leads to using an organized directory and file
+structure which can be reused across projects, reinforcing best practices,
+simplifying the process to ramp up new teammates, and making maintainable
+choices [_(StackOverflow by haylem)_][BBTSO].
```
Comment (generalredneck) [inline 2013-03-aquifer-and-build-systems/README.md]: Something about ease in debugging builds across environments... (maybe this is what "maintainable choices" is)

Maybe somethign about not consuming time running manual processes... DRY (don't repeat yourself)... why do something manually over and over again that you could script or automate.

File: 2013-03-aquifer-and-build-systems/README.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+[PCGH]: https://github.com/patrickocoffeyo
+[AQCDR]: https://github.com/aquifer/aquifer-coder
+[AQR]: https://github.com/aquifer/aquifer-run
+[AQQSG]: http://docs.aquifer.io/sections/quickstart-guide.html
+[NVM]: https://github.com/creationix/nvm
+[NJS]: https://nodejs.org/
+[NPMD]: https://github.com/mattgrill/NPM-Doctor
+[AQDS]: http://docs.aquifer.io/sections/directory-structure.html
+[AQG]: https://github.com/aquifer/aquifer-git
+[JTBC]: http://www.joelonsoftware.com/articles/fog0000000043.html
```
Comment (generalredneck) [inline 2013-03-aquifer-and-build-systems/README.md]: OMG... love you for URL references :+1:

Comment (milsyobtaf) [inline 2013-03-aquifer-and-build-systems/README.md]: Seriously. I didn't realize you could footnote your links and have it.... just work. :mind-blown:

**Pull request merged**